site/docs/developer_documentation/ONV_path_manipulation.md

```diff
@@ -0,0 +1,212 @@
+---
+id: developer_ONV_path
+title: ONV path manipulation
+sidebar_label: ONV path manipulation
+---
+
+Bit string manipulations can be very hard to understand. This section should help aspiring developers how GQCP deals with the manipulation of ONV's (occupation number vectors) and ONV paths. It is necessary that you understand the theory first, since the code tries to follow the theory as closely as possible. To brush up on CI algorithms and ONV manipulation we highly recommend chapter 11 in [Molecular Electronic‐Structure Theory](https://onlinelibrary.wiley.com/doi/book/10.1002/9781119019572) by Helgaker, Jørgensen and Olsen.
+
+## The key operators
+
+ONV path manipulations are described using a second quantization approach. GQCP's API sticks as closely to this as possible. To manipulate a path, we use the elementary second quantization operators to annihilate or create arcs that represent occupied orbitals. In an ONV path, a diagonal arc represents an occupied orbital, while a vertical arc represents an unoccupied orbital.  
+
+```C++
+/**
+*  Annihilate the diagonal arc that starts at the coordinate (q,n).
+* 
+*  @param q        the index of the orbital that should be annihilated
+*  @param n        the number of electrons in the ONV/path up to the orbital index q
+*/
+void ONVPath::annihilate(const size_t q, const size_t n) {
+
+    // During annihilation, we're removing an arc, so we'll have to update the current address by removing the corresponding arc weight.
+    this->m_address -= this->onv_basis.arcWeight(q, n);
+
+    // Update the next possible creation index. Since we're always constructing paths from the top-left to the bottom-right, we're only considering creation indices p > q.
+    this->p = q + 1;
+}
+
+/**
```

Comment:
I would separate `create` and `annihilate` into two separate code tabs and explain them separately.

Reply:
yep, that would be better indeed.